File: frontend/aktInstituteOS/src/types/admission.ts
export type AdmissionStatus =
  | "PENDING"
  | "DOCUMENTS_PENDING"
  | "ENROLLED"
  | "ACTIVE"
  | "COMPLETED"
  | "CANCELLED";

export interface Admission {
  id: number;
  uuid: string;
  admissionNumber: string;
  leadId: number;
  studentId?: number;
  firstName: string;
  lastName?: string;
  fullName: string;
  phone: string;
  email?: string;
  courseName?: string;
  batchId?: number;
  batchName?: string;
  feesAgreed: number;
  feesPaid: number;
  feesDue: number;
  enrollmentDate?: string;
  status: AdmissionStatus;
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

export interface AdmissionSummary {
  id: number;
  admissionNumber: string;
  leadId: number;
  studentId?: number;
  fullName: string;
  phone: string;
  courseName?: string;
  batchName?: string;
  feesAgreed: number;
  feesPaid: number;
  feesDue: number;
  enrollmentDate?: string;
  status: AdmissionStatus;
  createdAt: string;
}

export interface CreateAdmissionRequest {
  leadId: number;
  firstName: string;
  lastName?: string;
  phone: string;
  email?: string;
  courseName?: string;
  batchId?: number;
  batchName?: string;
  feesAgreed?: number;
  enrollmentDate?: string;
  notes?: string;
}

export interface UpdateAdmissionRequest {
  firstName?: string;
  lastName?: string;
  phone?: string;
  email?: string;
  courseName?: string;
  batchName?: string;
  feesAgreed?: number;
  feesPaid?: number;
  enrollmentDate?: string;
  notes?: string;
}

export interface EnrollStudentRequest {
  firstName: string;
  lastName?: string;
  phone: string;
  email?: string;
  whatsappNumber?: string;
  dateOfBirth?: string;
  gender?: "MALE" | "FEMALE" | "OTHER";
  address?: string;
  city?: string;
  state?: string;
  pincode?: string;
  parentName?: string;
  parentPhone?: string;
  parentEmail?: string;
  emergencyContact?: string;
  highestQualification?: string;
  schoolCollegeName?: string;
  aadhaarNumber?: string;
  panNumber?: string;
  notes?: string;
}

export interface AdmissionListParams {
  page?: number;
  size?: number;
  sort?: string;
  dir?: string;
  status?: AdmissionStatus | "";
  q?: string;
}

File: backend/aktInstituteOS/src/main/java/com/akt/institute/admission/domain/Admission.java
package com.akt.institute.admission.domain;

import com.akt.institute.shared.domain.BaseEntity;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Admission extends BaseEntity {

    private String uuid;
    private String admissionNumber;
    private Long instituteId;
    private Long leadId;
    private Long studentId;
    private String firstName;
    private String lastName;
    private String phone;
    private String email;
    private String courseName;
    private String batchName;
    private Long   batchId;     // FK to batches(id), optional
    @Builder.Default
    private BigDecimal feesAgreed = BigDecimal.ZERO;
    @Builder.Default
    private BigDecimal feesPaid = BigDecimal.ZERO;
    private LocalDate enrollmentDate;
    @Builder.Default
    private AdmissionStatus status = AdmissionStatus.PENDING;
    private String notes;

    public String getFullName() {
        return (lastName != null && !lastName.isBlank())
            ? firstName + " " + lastName
            : firstName;
    }

    public BigDecimal getFeesDue() {
        return feesAgreed.subtract(feesPaid);
    }
}
